JSDOMParser.js

```diff
@@ -1022,10 +1022,17 @@
             continue;
           }
           if (nextC === "/" && this.html.substr(this.currentChar, 8 /*"/script>".length */).toLowerCase() == "/script>") {
-            // Go back before the '<' so we find the end tag.
-            this.currentChar--;
-            // Done with this script tag, the caller will close:
-            return;
+            var indexOfNextScriptOpeningTag = this.html.indexOf("<script", this.currentChar);
+            var indexOfNextScriptClosingTag = this.html.indexOf("</script>", this.currentChar);
+            // Found out the closing script tag when there is no other opening/closing tag or
+            // index of closing tag is bigger than opening tag's, which means rest of opening/closing script tags are pairs.
```

Comment:
Unfortunately I also don't think it works well. Consider:

```
<script ...></script>
<!-- Something something </script> something -->
<p> random important content</p>
<script>
// more script
</script>
```

and now, if I'm reading your code right, the first closing of the `<script>` will be ignored because we spot a `</script>` lower down that is not preceded by a `<script>`.

I think the problem here is that the code tries to be HTML-style clever about how to read script tags, but if we're guaranteeing XHTML input that shouldn't be necessary and will actually cause problems in the case of nested script elements. What happens if you replace the call to `readScript` with `readChildren` ? Maybe we should just remove that code altogether. See also comments on #252